sdv/sequential/par.py

```diff
@@ -380,6 +380,10 @@ def _fit_sequence_columns(self, timeseries_data):
             kind = dtype.kind
             if kind in ('i', 'f'):
                 data_type = 'continuous'
+                # Check if metadata overrides this data type
```

Comment:
we can add a unit test that mocks `self._model` and makes sure it's called with categorical in the right index of data_types for this case